### ISSUE TITLE
Should dask be optional?

### ISSUE THREAD

**Issue by TomNicholas** [xgcm/xgcm#419]
In theory dask is meant to be optional for xgcm, but in practice we don't actually test that it works without dask installed in the CI.

Some of my changes in the grid ufunc refactor currently break it if dask isn't installed, but I think that if we decide we definitely do want it to work without dask then it's not a major fix.

Q is just: Do we want to bother with a no-dask option? (If so we should add a new CI env for it)

**Comment by jbusecke** [comment]
I think a non dask CI env would be a good idea, but I am not sure how much refactoring is needed in the tests? If you say its not a big deal, I support it.

Probably we actually need to modify the legacy tests to use dask at all?

**Comment by VeckoTheGecko** [comment]
Should also be quite straightforward with the work that https://github.com/lucascolley has done on the array API ecosystem in Python.

Reference:
- [Pydata Paris talk: Lucas Colley - A Hitchhiker's Guide to the Array API Standard Ecosystem](https://www.youtube.com/watch?v=LBVzkCSmjsQ) ([slides](https://lucascolley.github.io/talks/pydata-paris-25-array-api/#/6/2))
- [Array API compat library](https://data-apis.org/array-api-compat/)


Becomes easier with #683 as we can easily create envs without dask

I think I can work on this (and also take over/supercede #422 ) at some point

**Comment by TomNicholas** [comment]
I don't think that Lucas' work extends to chunked arrays. Computing on those requires using additional primitives such as `map_overlap` or `apply_gufunc`. But regardless I think this could be made to work fairly easily in xGCM.
